### PR TITLE
Andrew Kimmel hw 

### PR DESCRIPTION
**Pull request by adkimmel**

### CODE REVIEW

**Review by adkimmel — COMMENTED** [review]
A. Kimmel how submission

File: js/app.js [hunk 1]
```diff
@@ -24,6 +24,19 @@
...
+			throw new Error('Parameter type error.');
+		}
+		return {item, price};
+	};
+
+	const listItem1 = newShoppingListItem('milk', 3);
+	const listItem2 = newShoppingListItem('eggs', 1);
+	const listItem3 = newShoppingListItem('bacon', 7);
+	const listItem4 = newShoppingListItem('bagels', 4.95);
+	const listItem5 = newShoppingListItem('filet', 25);
```
Comment (sonylnagale) [inline js/app.js:38]: I presume the addition of items in your code throughout are to test your code? I would remove them before submitting, though your idea is sound.

File: js/app.js [hunk 2]
```diff
@@ -533,6 +640,12 @@
 
 	// implement function here
 
+	const combineLists = (items1, items2) => {
+		const temp = spliceItems(items1, -1);
+		spliceItems(items2, 99, temp);
```
Comment (sonylnagale) [inline js/app.js:645]: I think we can be more efficient here than hardcoding 99 into the code.

File: js/app.js [hunk 3]
```diff
@@ -763,6 +910,21 @@
...
+			throw new Error('i bigger than j')
+		}
+		if (i > list.length || j > list.length || i < 0 || j < 0) {
+			throw new Error('i/j out of range');
+		}
+		const subRange = list.splice(i,j - i + 1);
+		return computeSum(subRange);
+	};
+
+	const test = computeSumInRange(0,2,shoppingList);
```
Comment (sonylnagale) [inline js/app.js:925]: Can we use something that doesn't hardcode 0 and 2?

**Comment by sonylnagale** [comment]
Overall this is really good! I have a few suggestions that you can think about, but you don't have to resubmit.

**Pull request closed without merging**